spec/shared_context.rb

```diff
@@ -1,17 +1,19 @@
 # General context for unit testing:
 RSpec.shared_context 'shared context', a: :b do
   before :each do
-    options_120 = { url: 'https://oneview.example.com', user: 'Administrator', password: 'secret123', api_version: 120 }
-    @client_120 = OneviewSDK::Client.new(options_120)
-
-    options_200 = { url: 'https://oneview.example.com', user: 'Administrator', password: 'secret123' }
-    @client = OneviewSDK::Client.new(options_200)
-
-    options_300 = { url: 'https://oneview.example.com', user: 'Administrator', password: 'secret123', api_version: 300 }
-    @client_300 = OneviewSDK::Client.new(options_300)
+    options = { url: 'https://oneview.example.com', user: 'Administrator', password: 'secret123' }
+    # Creates dynamically the variables @client_120, @client_200 and etc.
+    api_versions = [120, 200, 300, 500]
```

Comment:
`api_versions = OneviewSDK::SUPPORTED_API_VERSIONS + [120]` 😉 

(and `OneviewSDK::ImageStreamer::SUPPORTED_API_VERSIONS` below for I3S)